runtime/libc/wasi_sdk_backing.c

```diff
@@ -0,0 +1,339 @@
+    u16 fs_flags;
+    u64 fs_rights_base;
+    u64 fs_rights_inheriting;
+};
+
+struct wasi_filestat {
+    u64 st_dev;
+    u64 st_ino;
+    u8 st_filetype;
+    u32 st_nlink;
+    u64 st_size;
+    u64 st_atim;
+    u64 st_mtim;
+    u64 st_ctim;
+};
+
+enum wasi_filetype {
+    WASI_FILETYPE_UNKNOWN           = 0,
+    WASI_FILETYPE_BLOCK_DEVICE      = 1,
+    WASI_FILETYPE_CHARACTER_DEVICE  = 2,
+    WASI_FILETYPE_DIRECTORY         = 3,
+    WASI_FILETYPE_REGULAR_FILE      = 4,
+    WASI_FILETYPE_SOCKET_DGRAM      = 5,
+    WASI_FILETYPE_SOCKET_STREAM     = 6,
+    WASI_FILETYPE_SYMBOLIC_LINK     = 7,
+};
+
+enum wasi_oflags {
+    WASI_O_CREAT        = 0x0001,
+    WASI_O_DIRECTORY    = 0x0002,
+    WASI_O_EXCL         = 0x0004,
+    WASI_O_TRUNC        = 0x0008,
+};
+
+enum wasi_fdflags {
+    WASI_FDFLAG_APPEND      = 0x0001,
+    WASI_FDFLAG_DSYNC       = 0x0002,
+    WASI_FDFLAG_NONBLOCK    = 0x0004,
+    WASI_FDFLAG_RSYNC       = 0x0008,
+    WASI_FDFLAG_SYNC        = 0x0010,
+};
+
+enum wasi_rights {
+    WASI_RIGHT_FD_DATASYNC              = 0x0000000000000001ULL,
+    WASI_RIGHT_FD_READ                  = 0x0000000000000002ULL,
+    WASI_RIGHT_FD_SEEK                  = 0x0000000000000004ULL,
+    WASI_RIGHT_FD_FDSTAT_SET_FLAGS      = 0x0000000000000008ULL,
+    WASI_RIGHT_FD_SYNC                  = 0x0000000000000010ULL,
+    WASI_RIGHT_FD_TELL                  = 0x0000000000000020ULL,
+    WASI_RIGHT_FD_WRITE                 = 0x0000000000000040ULL,
+    WASI_RIGHT_FD_ADVISE                = 0x0000000000000080ULL,
+    WASI_RIGHT_FD_ALLOCATE              = 0x0000000000000100ULL,
+    WASI_RIGHT_PATH_CREATE_DIRECTORY    = 0x0000000000000200ULL,
+    WASI_RIGHT_PATH_CREATE_FILE         = 0x0000000000000400ULL,
+    WASI_RIGHT_PATH_LINK_SOURCE         = 0x0000000000000800ULL,
+    WASI_RIGHT_PATH_LINK_TARGET         = 0x0000000000001000ULL,
+    WASI_RIGHT_PATH_OPEN                = 0x0000000000002000ULL,
+    WASI_RIGHT_FD_READDIR               = 0x0000000000004000ULL,
+    WASI_RIGHT_PATH_READLINK            = 0x0000000000008000ULL,
+    WASI_RIGHT_PATH_RENAME_SOURCE       = 0x0000000000010000ULL,
+    WASI_RIGHT_PATH_RENAME_TARGET       = 0x0000000000020000ULL,
+    WASI_RIGHT_PATH_FILESTAT_GET        = 0x0000000000040000ULL,
+    WASI_RIGHT_PATH_FILESTAT_SET_SIZE   = 0x0000000000080000ULL,
+    WASI_RIGHT_PATH_FILESTAT_SET_TIMES  = 0x0000000000100000ULL,
+    WASI_RIGHT_FD_FILESTAT_GET          = 0x0000000000200000ULL,
+    WASI_RIGHT_FD_FILESTAT_SET_SIZE     = 0x0000000000400000ULL,
+    WASI_RIGHT_FD_FILESTAT_SET_TIMES    = 0x0000000000800000ULL,
+    WASI_RIGHT_PATH_SYMLINK             = 0x0000000001000000ULL,
+    WASI_RIGHT_PATH_REMOVE_DIRECTORY    = 0x0000000002000000ULL,
+    WASI_RIGHT_PATH_UNLINK_FILE         = 0x0000000004000000ULL,
+    WASI_RIGHT_POLL_FD_READWRITE        = 0x0000000008000000ULL,
+    WASI_RIGHT_SOCK_SHUTDOWN            = 0x0000000010000000ULL,
+};
+
+
+// file operations
+i32 wasi_unstable_path_open(
```

Comment:
like the `_unstable_` namespacing.

Reply:
Wasn't me! This is the prefix WASI-SDK uses for its imports.